xbmc/drm/CryptoSession.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+/*
+ *      Copyright (C) 2005-2018 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Kodi; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <string>
+#include <vector>
+#include <map>
+#include <stdint.h>
+
+namespace DRM
+{
+  class CCryptoSession;
+
+  typedef CCryptoSession* (*GET_CRYPTO_SESSION_INTERFACE_FN)(const uint8_t *UUID, const std::string &cipherAlgo, const std::string &hmacAlgo);
+
+  class CCryptoSession
+  {
+  public:
+    // Interface registration
+    static CCryptoSession* GetCryptoSession(const uint8_t *UUID, const std::string &cipherAlgo, const std::string &macAlgo);
+    virtual ~CCryptoSession() {};
+
+    // Interface methods
+    virtual std::vector<char> GetKeyRequest(const std::vector<char> &init, const std::string &mimeType, bool offlineKey, const std::map<std::string, std::string> &parameters) = 0;
+    virtual std::string GetPropertyString(const std::string &name) = 0;
+    virtual std::string ProvideKeyResponse(const std::vector<char> &response) = 0;
+    virtual void RemoveKeys() = 0;
+    virtual void RestoreKeys(const std::string &keySetId) = 0;
+    virtual void SetPropertyString(const std::string &name, const std::string &value) = 0;
+
+    // Crypto methods
+    virtual std::vector<char> Decrypt(const std::vector<char> &cipherKeyId, const std::vector<char> &input, const std::vector<char> &iv) = 0;
+    virtual std::vector<char> Encrypt(const std::vector<char> &cipherKeyId, const std::vector<char> &input, const std::vector<char> &iv) = 0;
+    virtual std::vector<char> Sign(const std::vector<char> &macKeyId, const std::vector<char> &message) = 0;
+    virtual bool Verify(const std::vector<char> &macKeyId, const std::vector<char> &message, const std::vector<char> &signature ) = 0;
```

Comment:
If you're implementing these from scratch, why not use Buffer instead of std::vector<char>?